src/EnergyPlus/IndoorIceRink.hh

```diff
@@ -0,0 +1,287 @@
+        Real64 Qfusion;       // Enthalpy of water transformation to ice
+        Real64 Q;             // Final Q to be reported to EnergyPlus - using eqn. from low radiant temp system
+        Real64 SecInHour = 3600; 
+        Real64 Increments = 0.1667; 
+        Real64 TRefigOutCheck;     // Outlet temperature of refrigerant
+        Real64 RefrigTempIn;       // Inlet temperature of refrigerant
+        Real64 RefrigMassFlow;     // Refrigerant mass flow rate
+        Real64 PastRefrigMassFlow; // to keep track of previous mass flow rate
+        Real64 LoadMet;
+        Real64 COP = 2.5; // hardwired COP. This will be changed later.
+        Real64 IceSetPointTemp; //Ice rink setpoint
+        Real64 operation; //used to determine whether the ice rink is ON or OFF. 1: ON; 0: OFF
+        Real64 circuits = 423;
+        Real64 PipeArea;
+        Real64 EpsMdotCp;
+        
+
+        Real64 FreezingLoad;
+
+        // ReportData
+        Real64 RefrigInletTemp;  // Refrigerant inlet temperature
+        Real64 RefrigOutletTemp; // Refrigerant outlet temperature
+        Real64 CoolPower;        // Cooling sent to rink floor in Watts
+        Real64 CoolEnergy;       // Cooling sent to rink floor in Joules
+
+        // Members
+        std::string NameR; // Resurfacer Name
+        int CompIndex;
+        int WaterIndex;
+        std::string ResurfacingSchedName;
+        int ResurfacingSchedPtr;
+        int NoOfResurfEvents;
+        Real64 ResurfacingWaterTemp;
+        Real64 InitWaterTemp;
+        Real64 TankCapacity; // in m3
+
+        // Report Data
+        Real64 ResurfacingHeatLoad;
+        Real64 QResurfacing;
+        Real64 EHeatingWater;
+
+        // Function
+        Real64 RinkResurfacer(EnergyPlusData &state, Real64 &ResurfacingLoad);
+
+        // Default Constructor
+        IceRinkData()
+            : RinkType_Num(97), CompIndex(0), WaterIndex(0), ResurfacingSchedPtr(0), NoOfResurfEvents(0), ResurfacingWaterTemp(0.0),
+              InitWaterTemp(0.0), TankCapacity(0.0), QResurfacing(0.0), EHeatingWater(0.0), ResurfacingHeatLoad(0.0)
+        {
+        }
+        void
+        simulate(EnergyPlusData &state, const PlantLocation &calledFromLocation, bool FirstHVACIteration, Real64 &CurLoad, bool RunFlag) override;
+
+        void initialize(EnergyPlusData &state);
+
+        // void GetIndoorIceRink();
+
+        void setupOutputVariables(EnergyPlusData &state);
+
+        //Real64 PeopleHG(EnergyPlusData &state);
+
+        Real64 IceRinkFreezing(EnergyPlusData &state);
+
+        Real64 calcEffectiveness(EnergyPlusData &state, Real64 const Temperature, Real64 const RefrigMassFlow);
+
+        void calculateIceRink(EnergyPlusData &state, Real64 &LoadMet);
+
+        void update();
+
+        void report(bool RunFlag);
+
+        static PlantComponent *factory(EnergyPlusData &state, std::string const &objectName);
+    };
+
+    // Object Data:
+    extern Array1D<IceRinkData> Rink;
```

Comment:
All the non-const extern data needs to be moved into the `IceRinkData` class below and initialized in the `clear_state` method.